### ISSUE TITLE
Dependency Issues

### ISSUE THREAD

**Issue by JakobKlotz** [nasa/LHASA#26]
# Dependency Issues

Hi,

I'm on a Windows machine and tried to containerize the pfdf model with docker. I used the `pfdf/pfdf.yml` file to set up my environment, which installed the latest package versions. In turn these caused some issues.

## Minimal example

```bash
conda env create --file pfdf/pfdf.yml
source ~/.bashrc
conda activate pfdf
python pfdf/scripts/model_run.py --path=pfdf
```

Here is my `pfdf` environment with its package list:

[packages.txt](https://github.com/user-attachments/files/19206526/packages.txt)

---

A couple of issues that I faced with dependencies.

### `pyarrow` is missing

I missed `pyarrow` as dependency. Although from what I've seen the dependency `xarray-spatial` itself depends on `dask[dataframe]` ([ref](https://github.com/makepath/xarray-spatial/blob/6dd5faa2770e032dcc8c77eca71790051bd1d3ee/requirements.txt#L2)) which should contain `pyarrow` ([ref](https://github.com/dask/dask/blob/f20c0f2124fdcc4b8aa2fe6a74a72eefd96593ce/pyproject.toml#L53-L57)).

Simply installing `pyarrow` fixed the issue.

### `pandas` with `numpy.timedelta64(1, "M")`

`pandas` operations in conjunction with `numpy.timedelta64(1, "M")` are not working (at least with the version I'm running - `pandas==2.2.2`; [ref](https://github.com/pandas-dev/pandas/issues/52806)).

For a simple fix, I replaced the usage of `np.timedelta64(1,'M')` in these instances

[1](https://github.com/nasa/LHASA/blob/6cf8df47f61d17496a238d234fc6cd4d17af223c/pfdf/scripts/calc_basin_stats.py#L173) with:

```python
timedelta = np.ceil((today - pd.to_datetime(end)).days / 30.44) # just an approximation for months
```

and [2](https://github.com/nasa/LHASA/blob/6cf8df47f61d17496a238d234fc6cd4d17af223c/pfdf/scripts/firms_basin_join.py#L73) with:

```python
return_df['Duration'] = (return_df['FireEnd'] - return_df['FireStart']).days / 30.44 # just an approximation for months
```

... just to see, if I can get the model up and running.

### `pyogrio` has no header option

Instances were geojson files were written ([1](https://github.com/nasa/LHASA/blob/6cf8df47f61d17496a238d234fc6cd4d17af223c/pfdf/scripts/model_run.py#L228-L229), [2](https://github.com/nasa/LHASA/blob/6cf8df47f61d17496a238d234fc6cd4d17af223c/pfdf/scripts/calc_basin_stats.py#L234)), include `header=True` which lead to some issues, as later versions of `pyogrio` have no header option.
In my case, a removal of `header=True` did the trick.

---

Long story short, would you mind sharing further details on the package versions you're using within the `pfdf` environment?
Let me know, if I can be of further help, would be happy to contribute changes back to the project! 😄

Thank you!

**Comment by thomasastanley** [comment]
yes, i can see how the yml may not be working for you, as this was set up years ago now. In fact, I'm not sure if the pfdf module will work at this point, given that the online data archives may have changed. so before you invest a lot of time in it, you should be aware that you might also need to update the model code itself.

As for contributions, i'm happy to take a look at any changes you propose (such as the header issue above), but i can't guarantee a fast response.

**Comment by thomasastanley** [comment]
You can try rolling back to this environment:

channels:
  - conda-forge
  - defaults
dependencies:
  - _libgcc_mutex=0.1
  - _openmp_mutex=4.5
  - _py-xgboost-mutex=2.0
  - affine=2.4.0
  - aom=3.5.0
  - appdirs=1.4.4
  - attrs=23.1.0
  - beautifulsoup4=4.12.2
  - blosc=1.21.3
  - boltons=23.0.0
  - boost-cpp=1.78.0
  - boto3=1.26.126
  - botocore=1.29.126
  - branca=0.6.0
  - brotli=1.0.9
  - brotli-bin=1.0.9
  - brotlipy=0.7.0
  - brunsli=0.1
  - bs4=4.12.2
  - bzip2=1.0.8
  - c-ares=1.18.1
  - c-blosc2=2.8.0
  - ca-certificates=2023.5.7
  - cachetools=5.3.0
  - cairo=1.16.0
  - certifi=2023.5.7
  - cffi=1.15.1
  - cfitsio=4.2.0
  - cftime=1.6.2
  - charls=2.4.1
  - charset-normalizer=3.1.0
  - click=8.1.3
  - click-plugins=1.1.1
  - cligj=0.7.2
  - cloudpickle=2.2.1
  - colorama=0.4.6
  - colorcet=3.0.1
  - conda-package-handling=2.0.2
  - conda-package-streaming=0.8.0
  - contourpy=1.0.7
  - cryptography=40.0.2
  - curl=8.0.1
  - cycler=0.11.0
  - cytoolz=0.12.0
  - dask-core=2023.4.1
  - datashader=0.14.4
  - datashape=0.5.4
  - dav1d=1.0.0
  - expat=2.5.0
  - fiona=1.9.3
  - folium=0.14.0
  - font-ttf-dejavu-sans-mono=2.37
  - font-ttf-inconsolata=3.000
  - font-ttf-source-code-pro=2.038
  - font-ttf-ubuntu=0.83
  - fontconfig=2.14.2
  - fonts-conda-ecosystem=1
  - fonts-conda-forge=1
  - fonttools=4.39.3
  - freetype=2.12.1
  - freexl=1.0.6
  - fsspec=2023.4.0
  - gdal=3.6.3
  - geocube=0.4.0
  - geojson=3.0.1
  - geopandas=0.12.2
  - geopandas-base=0.12.2
  - geos=3.11.2
  - geotiff=1.7.1
  - gettext=0.21.1
  - giflib=5.2.1
  - hdf4=4.2.15
  - hdf5=1.12.2
  - icu=70.1
  - idna=3.4
  - imagecodecs=2023.1.23
  - imageio=2.28.1
  - importlib-metadata=6.6.0
  - importlib_metadata=6.6.0
  - jinja2=3.1.2
  - jmespath=1.0.1
  - joblib=1.2.0
  - json-c=0.16
  - jsonpatch=1.32
  - jsonpointer=2.0
  - jxrlib=1.1
  - kealib=1.5.0
  - keyutils=1.6.1
  - kiwisolver=1.4.4
  - krb5=1.20.1
  - lazy_loader=0.2
  - lcms2=2.15
  - ld_impl_linux-64=2.40
  - lerc=4.0.0
  - libaec=1.0.6
  - libavif=0.11.1
  - libblas=3.9.0
  - libbrotlicommon=1.0.9
  - libbrotlidec=1.0.9
  - libbrotlienc=1.0.9
  - libcblas=3.9.0
  - libcurl=8.0.1
  - libdeflate=1.17
  - libedit=3.1.20191231
  - libev=4.33
  - libexpat=2.5.0
  - libffi=3.4.2
  - libgcc-ng=12.2.0
  - libgdal=3.6.3
  - libgfortran-ng=12.2.0
  - libgfortran5=12.2.0
  - libglib=2.76.2
  - libgomp=12.2.0
  - libiconv=1.17
  - libjpeg-turbo=2.1.5.1
  - libkml=1.3.0
  - liblapack=3.9.0
  - libllvm11=11.1.0
  - libnetcdf=4.9.1
  - libnghttp2=1.52.0
  - libnsl=2.0.0
  - libopenblas=0.3.21
  - libpng=1.6.39
  - libpq=15.2
  - librttopo=1.1.0
  - libspatialindex=1.9.3
  - libspatialite=5.0.1
  - libsqlite=3.40.0
  - libssh2=1.10.0
  - libstdcxx-ng=12.2.0
  - libtiff=4.5.0
  - libuuid=2.38.1
  - libwebp-base=1.3.0
  - libxcb=1.13
  - libxgboost=1.7.4
  - libxml2=2.10.3
  - libzip=1.9.2
  - libzlib=1.2.13
  - libzopfli=1.0.3
  - llvmlite=0.39.1
  - locket=1.0.0
  - lz4-c=1.9.4
  - mapclassify=2.5.0
  - markupsafe=2.1.2
  - matplotlib-base=3.7.1
  - multipledispatch=0.6.0
  - munch=2.5.0
  - munkres=1.1.4
  - ncurses=6.3
  - netcdf4=1.6.3
  - networkx=3.1
  - noise=1.2.2
  - nspr=4.35
  - nss=3.89
  - numba=0.56.4
  - numpy=1.23.5
  - odc-geo=0.3.3
  - openjpeg=2.5.0
  - openssl=3.1.1
  - packaging=23.1
  - pandas=2.0.1
  - param=1.13.0
  - partd=1.4.0
  - pcre2=10.40
  - pillow=9.5.0
  - pip=23.1.2
  - pixman=0.40.0
  - planetary-computer=0.5.1
  - platformdirs=3.5.0
  - pluggy=1.2.0
  - pooch=1.7.0
  - poppler=23.03.0
  - poppler-data=0.4.12
  - postgresql=15.2
  - proj=9.1.1
  - pthread-stubs=0.4
  - py-xgboost=1.7.4
  - pycosat=0.6.4
  - pycparser=2.21
  - pyct=0.5.0
  - pyct-core=0.5.0
  - pydantic=1.10.7
  - pyopenssl=23.1.1
  - pyparsing=3.0.9
  - pyproj=3.5.0
  - pysheds=0.3.3
  - pysocks=1.7.1
  - pystac=1.7.3
  - pystac-client=0.6.1
  - python=3.10.10
  - python-dateutil=2.8.2
  - python-tzdata=2023.3
  - python_abi=3.10
  - pytz=2023.3
  - pywavelets=1.4.1
  - pyyaml=6.0
  - rasterio=1.3.6
  - readline=8.2
  - requests=2.29.0
  - rioxarray=0.14.1
  - rtree=1.0.1
  - ruamel.yaml=0.17.32
  - ruamel.yaml.clib=0.2.7
  - s3transfer=0.6.0
  - scikit-image=0.20.0
  - scikit-learn=1.2.2
  - scipy=1.10.1
  - setuptools=67.7.2
  - shapely=2.0.1
  - six=1.16.0
  - snappy=1.1.10
  - snuggs=1.4.7
  - soupsieve=2.3.2.post1
  - sqlite=3.40.0
  - stackstac=0.4.3
  - threadpoolctl=3.1.0
  - tifffile=2023.4.12
  - tiledb=2.13.2
  - tk=8.6.12
  - toolz=0.12.0
  - tqdm=4.65.0
  - typing-extensions=4.5.0
  - typing_extensions=4.5.0
  - tzcode=2023c
  - tzdata=2023c
  - unicodedata2=15.0.0
  - urllib3=1.26.15
  - wheel=0.40.0
  - xarray=2023.4.2
  - xarray-spatial=0.3.5
  - xerces-c=3.2.4
  - xgboost=1.7.4
  - xorg-kbproto=1.0.7
  - xorg-libice=1.0.10
  - xorg-libsm=1.2.3
  - xorg-libx11=1.8.4
  - xorg-libxau=1.0.9
  - xorg-libxdmcp=1.1.3
  - xorg-libxext=1.3.4
  - xorg-libxrender=0.9.10
  - xorg-renderproto=0.11.1
  - xorg-xextproto=7.3.0
  - xorg-xproto=7.0.31
  - xyzservices=2023.2.0
  - xz=5.2.6
  - yaml=0.2.5
  - zfp=1.0.0
  - zipp=3.15.0
  - zlib=1.2.13
  - zlib-ng=2.0.7
  - zstandard=0.19.0
  - zstd=1.5.2

**Comment by JakobKlotz** [comment]
Great, thanks for the environment file. Just came around to test it and it solved the issues I had with my environment! 
I'll have another shot at running the pfdf module later.
For now, I'll try to get the main model up and running on my Windows machine (might be interesting for #10). 😄

Thank you for the help!
